### ISSUE TITLE
Support HTTP/2 (& consequently gRPC)

### ISSUE THREAD

**Issue by liamwh** [oven-sh/bun#4414]
### What version of Bun is running?

0.8.1+16b4bf341acc0f4804f0b6bdf5298c180cd00366

### What platform is your computer?

Darwin 22.6.0 arm64 arm

### What steps can reproduce the bug?

Use gRPC with bufbuild connect. Some relevant packages include:

```json
    "@bufbuild/buf": "^1.26.1",
    "@bufbuild/protobuf": "^1.3.0",
    "@bufbuild/protoc-gen-connect-es": "^0.8.6",
    "@bufbuild/protoc-gen-es": "^1.3.0",
    "@bufbuild/connect": "^0.13.0",
    "@bufbuild/connect-node": "^0.13.0",
```

I am using this lib if you want to attempt to replicate unfortunately I cannot share source code here: https://github.com/aserto-dev/node-directory

Then running:

❯ bun --bun run dev
In my package.json, I have tried setting the dev command to be both of the following, but both crash:

```json
    "dev": "bunx --bun vite dev",
    "dev": "vite dev",
```

### What is the expected behavior?

The server doesn't crash.

### What do you see instead?

```
Internal server error: The "original" argument must be of type Function
      at promisify2 (node:util:365:99)
      at module code (/Users/myusernamehere/git/myproject/node_modules/.pnpm/@bufbuild+connect-node@0.13.0_@bufbuild+protobuf@1.3.0/node_modules/@bufbuild/con
nect-node/dist/esm/compression.js:27:31) (x3)
{"originalLine":365,"originalColumn":99}
error: "vite" exited with code 1 (SIGHUP)
error: script "dev" exited with code 1 (SIGHUP)
```

### Additional information

- Running with node my server does not crash.
- I have also tried using both SvelteKit adapter Node and SvelteKit adapter bun. With both the issue still occurs.

**Comment by paperclover** [comment]
can you open the second file in the stack trace, the one ending in `.../esm/compression.js`

and comment this line out
![image](https://github.com/oven-sh/bun/assets/24465214/546e0d8d-1349-4d97-b9de-4feb169b9773)

save and rebuild. promisify is throwing because the brotli compression isnt implemented at all and is `undefined`. if that is enough to get the build to work (as in, it doesnt ever try calling the compress and decompress function), we can define `brotliDecompress` as a function that throws just so this earlier error isnt hit.

otherwise, this will get resolved by #267

**Comment by utamori** [comment]
connect-node depends on http2 and bun doesn't support http/2 yet, so it won't work anyway

**Comment by liamwh** [comment]
Ok thanks, will stick with node until http/2 is supported. Is there any idea when that might be perhaps?

**Comment by aralroca** [comment]
> connect-node depends on http2 and bun doesn't support http/2 yet, so it won't work anyway

I also hope that Bun supports HTTP/2 soon

**Comment by thai-bug** [comment]
I really hope http2 will be soon implemented. Cause of gRPC

**Comment by Electroid** [comment]
We are working on this right now. Duplicate of #887
